operator/helper/bytesize.go

```diff
@@ -34,6 +34,14 @@ func (h *ByteSize) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return h.unmarshalShared(unmarshal)
 }
 
+func (h *ByteSize) UnmarshalText(text []byte) (err error) {
```

Comment:
I believe we should expand this test loop to also unmarshal the raw values using mapstructure, and validate the results are the same. 

Very rough implementation:

```
for i, tc := range cases {
		t.Run(strconv.Itoa(i), func(t *testing.T) {
			var bsViaYaml ByteSize
			err := yaml.Unmarshal([]byte(tc.input), &bsViaYaml)
			if tc.expectError {
				require.Error(t, err)
				return
			}

			require.NoError(t, err)
			require.Equal(t, tc.expected, bs)

                        // And also same result if using mapstructure
                        var bsViaMapstructure ByteSize
		        err := mapstructure.Decode([]byte(tc.input), &bsViaMapstructure)
                        if tc.expectError {
				require.Error(t, err)
				return
			}

			require.NoError(t, err)
			require.Equal(t, tc.expected, bs)
		})
	}
```